### PR TITLE
[1.1.x] Allow bed probe sanity checks to run

### PR DESCRIPTION
**Pull request by tcm0116**
#8554 packaged for 1.1.x

### CODE REVIEW

File: Marlin/SanityCheck.h [hunk 1]
```diff
@@ -616,7 +616,7 @@ static_assert(1 >= 0
   , "Please enable only one probe option: PROBE_MANUALLY, FIX_MOUNTED_PROBE, BLTOUCH, SOLENOID_PROBE, Z_PROBE_ALLEN_KEY, Z_PROBE_SLED, or Z Servo."
 );
 
-#if HAS_BED_PROBE
+#if PROBE_SELECTED && DISABLED(PROBE_MANUALLY)
```
Comment (thinkyhead) [inline Marlin/SanityCheck.h:619]: Maybe `HAS_BED_PROBE` can change instead…?
```diff
-#define PROBE_PIN_CONFIGURED (HAS_Z_MIN_PROBE_PIN || (HAS_Z_MIN && ENABLED(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN)))
-#define HAS_BED_PROBE (PROBE_SELECTED && PROBE_PIN_CONFIGURED && DISABLED(PROBE_MANUALLY))
+#define HAS_BED_PROBE (PROBE_SELECTED && DISABLED(PROBE_MANUALLY))
```

**Comment by tcm0116** [comment]
I haven't looked to see if there would be any side effects of changing
HAS_BED_PROBE.

**Comment by thinkyhead** [comment]
Looks like `HAS_BED_PROBE` is fine with the looser definition, and the `PROBE_PIN_CONFIGURED` conditions will be caught as part of the sanity check.

**Comment by tcm0116** [comment]
Are you doing to fix 2.0.x while you're at it?

**Comment by thinkyhead** [comment]
Already done, so… yes.

**Comment by tcm0116** [comment]
Quit being so efficient!

**Comment by thinkyhead** [comment]
So many little loose ends, I'm a-scared to release 1.1.7 because I know there will be 20 things waiting for us in the morning…

**Pull request closed without merging**